pdfminer/converter.py

```diff
@@ -723,3 +723,152 @@ def render(item: LTItem) -> None:
     def close(self) -> None:
         self.write_footer()
         return
+
+
+class HOCRConverter(PDFConverter):
+    """
+    Where text is being extracted from a variety of types of PDF within a
+    business process, those PDFs where the text is only present in image
+    form will need to be analysed using an OCR tool which will typically
+    output hOCR. This converter extracts the explicit text information from
+    those PDFs that do have it and uses it to genxerate a basic hOCR
+    representation that is designed to be used in conjunction with the image
+    of the PDF in the same way as genuine OCR output would be, but without the
+    inevitable OCR errors.
+
+    The converter does not handle images, diagrams or text colors.
+
+    In the examples processed by the contributor it was necessary to set
+    LAParams.all_texts to True.
+    """
+
+    CONTROL = re.compile(r"[\x00-\x08\x0b-\x0c\x0e-\x1f]")
+
+    def __init__(self, rsrcmgr, outfp, codec='utf8', pageno=1,
+                 laparams=None, stripcontrol=False):
+        PDFConverter.__init__(self, rsrcmgr, outfp, codec=codec, pageno=pageno,
+                              laparams=laparams)
+        self.stripcontrol = stripcontrol
+        self.within_chars = False
+        self.write_header()
+
+    def bbox_repr(self, bbox):
+        (x0, y0, x1, y1) = bbox
+        # pdf y-coordinates are the other way round from hOCR coordinates
+        return " ".join((
+            "bbox",
+            str(round(x0, 3)),
+            str(round(self.page_bbox[3]-y1, 0)),
+            str(round(x1, 0)),
+            str(round(self.page_bbox[3]-y0, 0))
+        ))
+
+    def write(self, text):
+        if self.codec:
+            text = text.encode(self.codec)
+        self.outfp.write(text)
+
+    def write_header(self):
+        if self.codec:
+            self.write(
+                "<html xmlns='http://www.w3.org/1999/xhtml' "
+                "xml:lang='en' lang='en' charset='%s'>\n" % self.codec)
```

Comment:
Preferably this would use a package / built-in library for formatting html. Preferably [`xml.etree.ElementTree`](https://docs.python.org/3/library/xml.etree.elementtree.html) because it does not require an extra dependency.

Reply:
ok